src/user.cpp

```diff
@@ -916,8 +916,6 @@ void neroshop::User::logout() {
     this->name.clear(); // clear name
     this->account_type = user_account_type::guest; // set account type to the default
     this->logged = false; // make sure user is no longer logged in
-    // delete this user
-    if(this) delete this;//this = nullptr;//fails
```

Comment:
You should not attempt to delete the class object instance from any of its member functions, including its destructor. This should be taken care of the calling code, either manually, if this was instantiated by a new/malloc (via delete/free) or, better, via a smart pointer, e.g., `std::unique_ptr`, which then deallocates memory automatically after (and whenever) the destructor is called. I'll look at the converting `new`s to 'std::unique_ptr's in a next round.